frontend/src/components/TodoTaskList/hook.ts

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from 'react';
+import { TodoTaskProps } from '../TodoTask/TodoTask';
+
+export const useTodoTasks = (todoTasks: TodoTaskProps[]) => {
+  const [tasks, setTasks] = useState<TodoTaskProps[]>([]);
+  const [completedTaskVisible, setCompletedTaskVisible] = useState(false);
+
+  const setIncompleteTasks = () => {
+    const incompleteTasks = todoTasks.filter((task) => task.isDone === false);
+    setTasks(incompleteTasks);
+  };
+
+  const handleCompleteTasks = () => {
```

Comment:
Nazwę tej funkcji musisz zmienić, bo długo rozkminiałem co znaczy ta nazwa.
Nir powinno być? Czy o to chodziło?
```suggestion
  const completeTask = () => {
```

Reply:
a może "show all tasks"?
a możę "handleCompleteTasksButtonClick"?
Tyko ta nazwa też nie odpowada super, bo raz ten button ma taką a raz taką nazwę.
Rozdzieliłbym to na dwie funkcje

Reply:
toggleTheVisibilityOfDoneTasks może być?